### PR TITLE
Feature/cross reference

### PR DESCRIPTION
**Pull request by ankitaShrma**
Updates on Previous code:
1. Take the numberFormat and convert to list number.
     Changes: - Create a new instance of createSFDT for each sfdt iteration where the stored renderedList is updated for same sfdt iteration but is cleared if computed for different sfdt.
Done:
- Process sfdt to generate the correct list number for the blocks
- Find the REF in block
- Find the correct ANCHOR for that REF from the updated list is 1st step
- Update the ref accordingly
TODO:
- Other num format except Arabic
Assumption: A list is only one Anchor
@p-mag @Anthony-Michel @hung-chou

### CODE REVIEW

**Comment by p-mag** [comment]
@ankitaShrma because of auto formatting, it hard to read.

Can you please do these things:
- add [prettier](https://prettier.io/) to `devDependencies` since we have [.prettierrc](https://github.com/zegal/sfdt-utils/blob/master/.prettierrc)
- add scripts in package.json `"fmt": prettier --config .prettierrc ./**.{ts,tsx,js,jsx}` (i dont remember the exact glob)
- update [README](https://github.com/zegal/sfdt-utils/blob/master/README) asking to run `yarn fmt` before commiting
- also run `yarn fmt` in your own branch

Also `dictionary.ts`, `listHelpers.ts`, we can use `syncfusion` official packages instead of forking theirs.

File: src/sfdt/listHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
+import {sfdt} from './../../types/sfdt.d';
+const {isNull, isUndefined, get, find} = require('lodash');
```
Comment (p-mag) [inline src/sfdt/listHelpers.ts]: Can we avoid mix matching of import styles and stick to one import styles?

File: src/dictionary.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import {isNullOrUndefined} from './sfdt/listHelpers';
+
```
Comment (p-mag) [inline src/dictionary.ts]: btw, If we want to make a true private members in javascript we should use closures.
> https://www.crockford.com/javascript/private.html

File: src/sfdt/blocksProcess.ts [hunk 3]
```diff
@@ -58,27 +81,27 @@ export const processBlock = (block: any, callbackInline: TCallback, callbackBloc
 //  - characterFormat
 export const processBlocks = (parent: any, callback: TCallback) => {
 	if (!parent.sections) {
-		// console.warn('Missing: sections', parent)
-		return false
+		console.warn('Missing: sections', parent);
```
Comment (p-mag) [inline src/sfdt/blocksProcess.ts]: can you please comment out this line? We dont want debug log in prod :)

Comment (ankitaShrma) [reply src/sfdt/blocksProcess.ts]: done

File: src/sfdt/crossReference.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+const REF = 'XREF:';
+
+const getAnchorName = (anchorBookmark) => anchorBookmark.split(ANCHOR)[1];
+const getRefName = (refBookMark) => refBookMark.split(REF)[1];
+
+const inlinesWithAnchor = (inlines) =>
+	doInlinesContain(inlines, (inline) => {
+		const name = get(inline, 'name');
+
+		const processedName = name; //&& name.split(':');
```
Comment (p-mag) [inline src/sfdt/crossReference.ts]: can you please do
`const processedName  = get(inline, 'name');`

* same for all other down the file.

File: src/sfdt/listHelpers.ts [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+// /**
+// * Gets list text
+// * @param listAdv
+// * @param listLevelNumber
+// * @param currentListLevel
+// * @param document
+// */
+// getListText(listAdv, listLevelNumber, currentListLevel) {
```
Comment (Anthony-Michel) [inline src/sfdt/listHelpers.ts]: please delete all these

**Pull request merged**